qa/tasks/cephfs/test_admin.py

```diff
@@ -194,6 +195,172 @@ def test_fs_new_pool_application_metadata(self):
+        self.mount_a.umount_wait()
+        self.run_cluster_cmd(f'auth rm client.{client_id}')
+        self.run_cluster_cmd(f'fs rename {new_fs_name} {orig_fs_name} --yes-i-really-mean-it')
+        self.fs.name = orig_fs_name
+
+    def test_fs_rename_idempotency(self):
+        """
+        That the file system rename operation is idempotent.
+        """
+        # Renaming the file system breaks this mount as the client uses
+        # file system specific authorization.
+        self.mount_a.umount_wait(require_clean=True)
+        orig_fs_name = self.fs.name
+        new_fs_name = 'new_cephfs'
+
+        self.run_cluster_cmd(f'fs rename {orig_fs_name} {new_fs_name} --yes-i-really-mean-it')
+        self.run_cluster_cmd(f'fs rename {orig_fs_name} {new_fs_name} --yes-i-really-mean-it')
+
+        # original file system name does not appear in `fs ls` command
+        self.assertFalse(self.fs.exists())
+        self.fs.name = new_fs_name
+        self.assertTrue(self.fs.exists())
+
+        # cleanup
+        self.run_cluster_cmd(f'fs rename {new_fs_name} {orig_fs_name} --yes-i-really-mean-it')
+        self.fs.name = orig_fs_name
+
+    def test_fs_rename_fs_new_fails_with_old_fsname_existing_pools(self):
+        """
+        That after renaming a file system, creating a file system with
+        old name and existing FS pools fails.
+        """
+        # Renaming the file system breaks this mount as the client uses
+        # file system specific authorization.
+        self.mount_a.umount_wait(require_clean=True)
+        orig_fs_name = self.fs.name
+        new_fs_name = 'new_cephfs'
+        data_pool = self.fs.get_data_pool_name()
+        metadata_pool = self.fs.get_metadata_pool_name()
+        self.run_cluster_cmd(f'fs rename {orig_fs_name} {new_fs_name} --yes-i-really-mean-it')
+
+        try:
+            self.run_cluster_cmd(f"fs new {orig_fs_name} {metadata_pool} {data_pool}")
+        except CommandFailedError as ce:
+            self.assertEqual(ce.exitstatus, errno.EINVAL,
+                "invalid error code on creating a new file system with old "
+                "name and existing pools.")
+        else:
+            self.fail("expected creating new file system with old name and "
+                      "existing pools to fail.")
+
+        try:
+            self.run_cluster_cmd(f"fs new {orig_fs_name} {metadata_pool} {data_pool} --force")
+        except CommandFailedError as ce:
+            self.assertEqual(ce.exitstatus, errno.EEXIST,
+                "invalid error code on creating a new file system with old "
+                "name, existing pools and --force flag.")
+        else:
+            self.fail("expected creating new file system with old name, "
+                      "existing pools, and --force flag to fail.")
+
+        try:
+            self.run_cluster_cmd(f"fs new {orig_fs_name} {metadata_pool} {data_pool} "
+                                 "--allow-dangerous-metadata-overlay")
```

Comment:
This one fails too?

Reply:
Yes. You need '--force' flag in addition to '--allow-dangerous-metadata-overlay' for this to work.